### PR TITLE
Ported window-margins

### LINKED ISSUE #343: Port window-margins to Rust
```c
DEFUN ("window-margins", Fwindow_margins, Swindow_margins,
       0, 1, 0,
       doc: /* Get width of marginal areas of window WINDOW.
WINDOW must be a live window and defaults to the selected one.

Value is a cons of the form (LEFT-WIDTH . RIGHT-WIDTH).
If a marginal area does not exist, its width will be returned
as nil.  */)
  (Lisp_Object window)
{
  struct window *w = decode_live_window (window);
  return Fcons (w->left_margin_cols
		? make_number (w->left_margin_cols) : Qnil,
		w->right_margin_cols
		? make_number (w->right_margin_cols) : Qnil);
}
```

### PR DESCRIPTION
**Pull request by Yarikx**
Closes #343

### CODE REVIEW

File: rust_src/src/windows.rs [hunk 1]
```diff
@@ -135,3 +135,32 @@ pub fn window_minibuffer_p(window: LispObject) -> LispObject {
...
+/// If a marginal area does not exist, its width will be returned
+/// as nil.
+#[lisp_fn(min = "0")]
+pub fn window_margins(window: LispObject) -> LispObject {
+    let win = if window.is_nil() {
+        selected_window()
+    } else {
+        window
+    }.as_window_or_error();
+
```
Comment (brotzeit) [inline rust_src/src/windows.rs]: It has to be a live window.

File: rust_src/src/windows.rs [hunk 2]
```diff
@@ -135,3 +135,32 @@ pub fn window_minibuffer_p(window: LispObject) -> LispObject {
...
+/// as nil.
+#[lisp_fn(min = "0")]
+pub fn window_margins(window: LispObject) -> LispObject {
+    let win = if window.is_nil() {
+        selected_window()
+    } else {
+        window
+    }.as_live_window_or_error();
+
+    let left = if win.left_margin_cols != 0 {
```
Comment (shanavas786) [inline rust_src/src/windows.rs]: An inner function would remove code duplication.

**Comment by Wilfred** [comment]
@Yarikx I've added you as a collaborator, so you can also review PRs if you wish :)

**Comment by Yarikx** [comment]
@wilfred Thanks! I'll do my best to help this project and community

**Pull request merged**